crud/common/utils.lua

```diff
@@ -199,6 +199,84 @@ function utils.tarantool_supports_uuids()
     return enabled_tarantool_features.uuids
 end
 
+local function map_to_table(map)
+    local tab = {}
+    for _, value in pairs(map) do
+        table.insert(tab, value)
+    end
+
+    return tab
+end
+
+local function add_nullable_fields_recursive(operations, space_format, tuple, id)
+    if id < 2 or tuple[id - 1] ~= box.NULL then
+        return operations
+    end
+
+    if space_format[id - 1].is_nullable and not operations[id - 1] then
+        operations[id - 1] = {'=', id - 1, box.NULL}
+        return add_nullable_fields_recursive(operations, space_format, tuple, id - 1)
+    end
+
+    return operations
+end
+
+-- Tarantool < 2.3 has no fields `box.error.NO_SUCH_FIELD_NO` and `box.error.NO_SUCH_FIELD_NAME`.
+if _TARANTOOL >= "2.3" then
+    function utils.is_field_not_found(err_code)
+        return err_code == box.error.NO_SUCH_FIELD_NO or err_code == box.error.NO_SUCH_FIELD_NAME
+    end
+else
+    function utils.is_field_not_found(err_code)
+        return err_code == box.error.NO_SUCH_FIELD
+    end
+end
+
+function utils.add_intermediate_nullable_fields(operations, space_format, tuple)
+    if tuple == nil then
+        return operations
+    end
+
+    -- We need this map to check existence of an operation in constant complexity
+    local operations_map, err = utils.get_operations_map(operations, space_format)
+    if err ~= nil then
+        return operations
+    end
+
+    for key, _ in pairs(operations_map) do
+        operations_map = add_nullable_fields_recursive(
+            operations_map, space_format,
+            tuple, key
+        )
+    end
+
+    return map_to_table(operations_map)
+end
+
+local function convert_jsonpath_operation(operation)
+    local field_parts = operation[2]:split('.', 1)
+    local field_name = field_parts[1]
+    local field_id = tonumber(field_name:sub(2, -2))
+
+    if #field_parts > 1 or field_id == nil then
+        -- Checking '[4].a.b' case
+        if field_name:sub(1, 1) == '[' and field_name:sub(-1, -1) == ']' then
+            field_id = tonumber(field_name:sub(2, -2))
+
+            if field_id == nil then
+                -- Checking '["field_name"]' case
+                if field_name:sub(2, 2) == '"' and field_name:sub(-2, -2) == '"' then
+                    return field_name:sub(3, -3), nil
+                end
+            end
+
+            return nil, tonumber(field_name:sub(2, -2))
+        end
+    end
+
+    return field_name, nil
+end
```

Comment:
Please don't try to parse jsonpaths. You could see json parser that compatible with Tarantool jsonpath here - https://github.com/olegrok/tarantool-jsonpath/blob/master/jsonpath.lua

Actually, you don't need here